### PR TITLE
HDDS-1488. Scm cli command to start/stop replication manager.

### PR DESCRIPTION
**Pull request by nandakumar131**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 145 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 50 | Maven dependency ordering for branch |
| +1 | mvninstall | 801 | trunk passed |
| +1 | compile | 427 | trunk passed |
| +1 | checkstyle | 84 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1044 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 204 | trunk passed |
| 0 | spotbugs | 528 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 754 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for patch |
| +1 | mvninstall | 601 | the patch passed |
| +1 | compile | 434 | the patch passed |
| +1 | cc | 434 | the patch passed |
| +1 | javac | 434 | the patch passed |
| -0 | checkstyle | 39 | hadoop-hdds: The patch generated 7 new + 0 unchanged - 0 fixed = 7 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 725 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 165 | the patch passed |
| -1 | findbugs | 211 | hadoop-hdds generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
||| _ Other Tests _ |
| +1 | unit | 325 | hadoop-hdds in the patch passed. |
| -1 | unit | 2440 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 9191 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-hdds |
|  |  Inconsistent synchronization of org.apache.hadoop.hdds.scm.container.ReplicationManager.replicationMonitor; locked 57% of time  Unsynchronized access at ReplicationManager.java:57% of time  Unsynchronized access at ReplicationManager.java:[line 169] |
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.om.TestSecureOzoneManager |
|   | hadoop.hdds.scm.pipeline.TestPipelineClose |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1221 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux b631570a5548 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 8f40856 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/1/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/1/artifact/out/new-findbugs-hadoop-hdds.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/1/testReport/ |
| Max. process+thread count | 5243 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/client hadoop-hdds/server-scm hadoop-hdds/tools U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java [hunk 1]
```diff
@@ -469,6 +469,27 @@ public boolean forceExitSafeMode() throws IOException {
...
+    AUDIT.logWriteSuccess(buildAuditMessageForSuccess(
+        SCMAction.START_REPLICATION_MANAGER, null));
+    scm.getReplicationManager().start();
+  }
+
+  @Override
+  public void stopReplicationManager() {
+    AUDIT.logWriteSuccess(buildAuditMessageForSuccess(
+        SCMAction.STOP_REPLICATION_MANAGER, null));
+    scm.getReplicationManager().stop();
```
Comment (bharatviswa504) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: Shall we want to return a boolean flag which shows whether the stop is a success or not?
Like case calling stop on already stopped replication monitor, we can print like Replication Monitor is not running, to stop it(Or some message like this). 

Same applicable for start also.

Comment (nandakumar131) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: Stop is an idempotent call and also it's async, so getting status of ReplicationManager immediately after calling stop might some time return true (running).

> Like case calling stop on already stopped replication monitor, we can print like Replication Monitor is not running

If we do this, we can end up in a minor inconsistency for a short period of time. Where stop will say that ReplicationManager is not running, but the status command will say that ReplicationManager is running.

Comment (bharatviswa504) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: This is true, even if we don't return status, as stop() is not a blocking call.
Can we print some output message for the command for stop like we signalled for stop, and if immediately status is called you can still see it as running. (Or something more clearer, to let users know the behavior)

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java [hunk 2]
```diff
@@ -154,6 +151,9 @@ public synchronized void start() {
     if (!running) {
```
Comment (bharatviswa504) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java]: Here we need to check isRunning() method.
Take an example:
call start, stop, (but it has not stopped completely, calling start immediately we try to start again if we only check running flag.

Comment (nandakumar131) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java]: Makes sense, will change it.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java (hunk 1 above)
Comment (nandakumar131) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: Sure.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 85 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for branch |
| +1 | mvninstall | 612 | trunk passed |
| +1 | compile | 376 | trunk passed |
| +1 | checkstyle | 73 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 957 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 215 | trunk passed |
| 0 | spotbugs | 539 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 786 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for patch |
| +1 | mvninstall | 687 | the patch passed |
| +1 | compile | 456 | the patch passed |
| +1 | cc | 456 | the patch passed |
| +1 | javac | 456 | the patch passed |
| -0 | checkstyle | 41 | hadoop-hdds: The patch generated 7 new + 0 unchanged - 0 fixed = 7 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 853 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 198 | the patch passed |
| -1 | findbugs | 269 | hadoop-hdds generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0) |
||| _ Other Tests _ |
| +1 | unit | 406 | hadoop-hdds in the patch passed. |
| -1 | unit | 3308 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 61 | The patch does not generate ASF License warnings. |
| | | 10198 | |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:hadoop-hdds |
|  |  Inconsistent synchronization of org.apache.hadoop.hdds.scm.container.ReplicationManager.replicationMonitor; locked 57% of time  Unsynchronized access at ReplicationManager.java:57% of time  Unsynchronized access at ReplicationManager.java:[line 169] |
| Failed junit tests | hadoop.ozone.TestMiniChaosOzoneCluster |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.client.rpc.Test2WayCommitInRatis |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.hdds.scm.pipeline.TestSCMPipelineManager |
|   | hadoop.ozone.om.TestOzoneManager |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerByPipeline |
|   | hadoop.ozone.om.TestOMRatisSnapshots |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.hdds.scm.pipeline.TestPipelineClose |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1221 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux f5088818da38 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 71aad60 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/2/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/2/artifact/out/new-findbugs-hadoop-hdds.html |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/2/testReport/ |
| Max. process+thread count | 3493 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/client hadoop-hdds/server-scm hadoop-hdds/tools U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 42 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 583 | trunk passed |
| +1 | compile | 365 | trunk passed |
| +1 | checkstyle | 74 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 830 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 159 | trunk passed |
| 0 | spotbugs | 422 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 613 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for patch |
| +1 | mvninstall | 546 | the patch passed |
| +1 | compile | 358 | the patch passed |
| +1 | cc | 358 | the patch passed |
| +1 | javac | 358 | the patch passed |
| +1 | checkstyle | 79 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 650 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 150 | the patch passed |
| +1 | findbugs | 626 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 318 | hadoop-hdds in the patch passed. |
| -1 | unit | 2394 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 8052 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestContainerStateMachineFailures |
|   | hadoop.ozone.client.rpc.TestOzoneAtRestEncryption |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.client.rpc.TestContainerStateMachine |
|   | hadoop.hdds.scm.pipeline.TestRatisPipelineProvider |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |
|   | hadoop.hdds.scm.pipeline.TestPipelineClose |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1221 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 0a3c21a34f04 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d6697da |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/3/testReport/ |
| Max. process+thread count | 4793 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/client hadoop-hdds/server-scm hadoop-hdds/tools U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 49 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 617 | trunk passed |
| +1 | compile | 395 | trunk passed |
| +1 | checkstyle | 77 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 865 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 168 | trunk passed |
| 0 | spotbugs | 469 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 707 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 18 | Maven dependency ordering for patch |
| +1 | mvninstall | 558 | the patch passed |
| +1 | compile | 384 | the patch passed |
| +1 | cc | 384 | the patch passed |
| +1 | javac | 384 | the patch passed |
| +1 | checkstyle | 69 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 686 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 174 | the patch passed |
| +1 | findbugs | 734 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 299 | hadoop-hdds in the patch passed. |
| -1 | unit | 2091 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 8102 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdds.scm.pipeline.TestRatisPipelineCreateAndDestory |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.hdds.scm.pipeline.TestPipelineClose |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1221 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 47992829ceda 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d6697da |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/5/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/5/testReport/ |
| Max. process+thread count | 5341 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/client hadoop-hdds/server-scm hadoop-hdds/tools U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: I just see it is mentioned Stopping ReplicationManager... With this info, the user will not understand this is async behavior and immediately calling getStatus he might still see it is running.

**Comment by bharatviswa504** [comment]
One suggestion, regarding output message for the commands.
Rest LGTM.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java (hunk 1 above)
Comment (nandakumar131) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java]: Updated the message.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for branch |
| +1 | mvninstall | 591 | trunk passed |
| +1 | compile | 371 | trunk passed |
| +1 | checkstyle | 73 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 838 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 172 | trunk passed |
| 0 | spotbugs | 430 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 629 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 18 | Maven dependency ordering for patch |
| +1 | mvninstall | 548 | the patch passed |
| +1 | compile | 378 | the patch passed |
| +1 | cc | 378 | the patch passed |
| +1 | javac | 378 | the patch passed |
| +1 | checkstyle | 80 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 677 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 157 | the patch passed |
| +1 | findbugs | 646 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 285 | hadoop-hdds in the patch passed. |
| -1 | unit | 167 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 42 | The patch does not generate ASF License warnings. |
| | | 5902 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.ratis.TestOzoneManagerDoubleBufferWithOMResponse |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1221 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux af5370dc1858 4.4.0-157-generic #185-Ubuntu SMP Tue Jul 23 09:17:01 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1127215 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/6/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/6/testReport/ |
| Max. process+thread count | 504 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/client hadoop-hdds/server-scm hadoop-hdds/tools U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by bharatviswa504** [comment]
+1 LGTM.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 93 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 42 | Maven dependency ordering for branch |
| +1 | mvninstall | 692 | trunk passed |
| +1 | compile | 369 | trunk passed |
| +1 | checkstyle | 70 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 885 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 158 | trunk passed |
| 0 | spotbugs | 444 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 674 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 18 | Maven dependency ordering for patch |
| +1 | mvninstall | 557 | the patch passed |
| +1 | compile | 376 | the patch passed |
| +1 | cc | 376 | the patch passed |
| +1 | javac | 376 | the patch passed |
| +1 | checkstyle | 90 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 726 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 159 | the patch passed |
| -1 | findbugs | 379 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| +1 | unit | 337 | hadoop-hdds in the patch passed. |
| -1 | unit | 459 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 39 | The patch does not generate ASF License warnings. |
| | | 6490 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1221 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 09e331291846 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3cc0ace |
| Default Java | 1.8.0_212 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/7/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/7/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/7/testReport/ |
| Max. process+thread count | 1197 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/client hadoop-hdds/server-scm hadoop-hdds/tools U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 51 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for branch |
| +1 | mvninstall | 633 | trunk passed |
| +1 | compile | 376 | trunk passed |
| +1 | checkstyle | 65 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 856 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 178 | trunk passed |
| 0 | spotbugs | 493 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 735 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 19 | Maven dependency ordering for patch |
| +1 | mvninstall | 574 | the patch passed |
| +1 | compile | 390 | the patch passed |
| +1 | cc | 390 | the patch passed |
| +1 | javac | 390 | the patch passed |
| +1 | checkstyle | 72 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 668 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 159 | the patch passed |
| +1 | findbugs | 704 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 208 | hadoop-hdds in the patch failed. |
| -1 | unit | 1737 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 38 | The patch does not generate ASF License warnings. |
| | | 7658 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.statemachine.commandhandler.TestCloseContainerCommandHandler |
|   | hadoop.ozone.client.rpc.TestBlockOutputStreamWithFailures |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmSafeMode |
|   | hadoop.ozone.om.TestKeyManagerImpl |
|   | hadoop.ozone.client.rpc.TestOzoneRpcClient |
|   | hadoop.ozone.scm.pipeline.TestSCMPipelineMetrics |
|   | hadoop.ozone.om.TestOzoneManagerHA |
|   | hadoop.ozone.client.rpc.TestMultiBlockWritesWithDnFailures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1221 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle cc |
| uname | Linux 3b196b706a16 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 00b5a27 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/8/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/8/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/8/testReport/ |
| Max. process+thread count | 5336 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/common hadoop-hdds/client hadoop-hdds/server-scm hadoop-hdds/tools U: hadoop-hdds |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1221/8/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**